app/uk/gov/hmrc/cataloguefrontend/search/IndexBuilder.scala

```diff
@@ -88,12 +88,19 @@ object IndexBuilder {
 
   // TODO: we could cache the results short term, generally the next query will be the previous query + 1 letter
   //       so we can reuse the partial result set
-  def search(query: Seq[String], index: Seq[SearchTerm]): Seq[SearchTerm] = {
-      query
-        .map(normalizeTerm)
-        .foldLeft(index) {
-          case (acc, cur) => acc.filter(_.terms.exists(_.contains(cur)))
-        }.sortWith( (a,b) => a.weight > b.weight )
+
+  def search(query: Seq[String], index: Map[String, Seq[SearchTerm]]): Seq[SearchTerm] = {
```

Comment:
is the `index` passed in for tests?
If so, I'd be tempted to make this a package private function, and then add `def search(query: Seq[String]) = search(query, cachedIndex)` as the public API

Reply:
Actually, I wonder how many other functions provided by `IndexBuilder` are really internal. And is the name `IndexBuilder` appropriate, if it's main public function is `search` - maybe it's just `SearchIndex`?